### PR TITLE
fix(evaluate): fix evaluate dict merge for pydantic

### PR DESCRIPTION
**Pull request by MSNP1381**
it had error when your module's output was a pydantic object rather that a dict

### CODE REVIEW

File: dsp/modules/gpt3.py [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
 import backoff
 import httpx
-try:
+
+from dsp.modules.cache_utils import CacheMemory, NotebookCacheMemory, cache_turn_on
+from dsp.modules.lm import LM
+from dsp.utils.settings import settings
+
+
+def is_module_installed(module_name):
```
Comment (arnavsinghvi11) [inline dsp/modules/gpt3.py:16]: just curious why this is needed. does the try/except block not take care of this check if langfuse is not supported?

File: dsp/modules/gpt3.py [hunk 2]
```diff
@@ -111,6 +122,11 @@ def __init__(
     def _openai_client(self):
         return openai
 
+    # TODO: add test suite to me.
```
Comment (arnavsinghvi11) [inline dsp/modules/gpt3.py:125]: remove extra code

**Review by arnavsinghvi11 — CHANGES_REQUESTED** [review]
Thanks @MSNP1381 ! left some minor comments to avoid extraneous changes to`gpt3.py` but the changes to `evaluate` LGTM

**Comment by MSNP1381** [comment]
> Thanks @MSNP1381 ! left some minor comments to avoid extraneous changes to`gpt3.py` but the changes to `evaluate` LGTM

i just added the code comments but the merge function is unstable be cause there is varying type of output for modules. My opinion is either we just use pydantic Object or any inherited objects or user have to follow some basic interfaces (as simple as dict interface))

**Comment by MSNP1381** [comment]
@okhat can i talk to you through the email?

**Pull request closed without merging**